lms/djangoapps/discussion_api/api.py

```diff
@@ -276,7 +286,15 @@ def get_thread_list(request, course_key, page, page_size, topic_id_list=None, te
         "per_page": page_size,
         "text": text_search,
     }
+
     text_search_rewrite = None
+
+    if view:
+        if view in ["unread", "unanswered"]:
+            query_params[view] = "true"
+        else:
+            HttpResponseBadRequest("Invalid value for view: {}".format(view))
```

Comment:
I'm confused why an api method is throwing an http error. It seems inconsistent with an API method and what is returned in the non-error case.